modules/reindex/src/internalClusterTest/java/org/elasticsearch/index/reindex/CrossClusterReindexIT.java

```diff
@@ -0,0 +1,166 @@
+            client.prepareIndex(index).setSource("f", "v").get();
+        }
+        client.admin().indices().prepareRefresh(index).get();
+        return numDocs;
+    }
+
+    public void testReindexFromRemote_IndexExists_success() throws Exception {
+        assertAcked(client(REMOTE_CLUSTER).admin().indices().prepareCreate("source-index-001"));
+        final int docsNumber = indexDocs(client(REMOTE_CLUSTER), "source-index-001");
+
+        final String sourceIndexInRemote = REMOTE_CLUSTER + ":" + "source-index-001";
+        new ReindexRequestBuilder(client(LOCAL_CLUSTER), ReindexAction.INSTANCE).source(sourceIndexInRemote)
+            .destination("desc-index-001")
+            .ignoreUnavailable(true)
+            .get();
+
+        assertTrue("Number of documents in source and desc indexes does not match", waitUntil(() -> {
+            SearchResponse resp = client(LOCAL_CLUSTER).prepareSearch("desc-index-001")
+                .setQuery(new MatchAllQueryBuilder())
+                .setSize(1000)
+                .get();
+            final TotalHits totalHits = resp.getHits().getTotalHits();
+            return totalHits.relation == TotalHits.Relation.EQUAL_TO && totalHits.value == docsNumber;
+        }));
+    }
+
+    public void testReindexFromRemote_IndexNotExists_success() throws Exception {
+
+        BulkByScrollResponse response = new ReindexRequestBuilder(client(LOCAL_CLUSTER), ReindexAction.INSTANCE).source(
+            REMOTE_CLUSTER + ":" + "no-such-source-index-001"
+        ).destination("desc-index-001").ignoreUnavailable(true).get();
+
+        // assert that nothing was copied over
+        assertThat(response.getTotal(), equalTo(0L));
+        assertThat(response.getUpdated(), equalTo(0L));
+        assertThat(response.getCreated(), equalTo(0L));
+        assertThat(response.getDeleted(), equalTo(0L));
+
+        // assert that local index was not created either
+        final IndexNotFoundException e = expectThrows(
+            IndexNotFoundException.class,
+            () -> client(LOCAL_CLUSTER).prepareSearch("desc-index-001").setQuery(new MatchAllQueryBuilder()).setSize(1000).get()
+        );
+
+        assertThat(e, hasToString(containsString(String.format(Locale.ROOT, "no such index [%s]", "desc-index-001"))));
+    }
+
+    public void testReindexFromRemote_CrossClusterCallsNotSupportedError() throws Exception {
+        assertAcked(client(REMOTE_CLUSTER).admin().indices().prepareCreate("source-index-001"));
+        final int docsNumber = indexDocs(client(REMOTE_CLUSTER), "source-index-001");
+
+        final String sourceIndexInRemote = REMOTE_CLUSTER + ":" + "source-index-001";
+
+        final IllegalArgumentException e = expectThrows(
+            IllegalArgumentException.class,
+            () -> new ReindexRequestBuilder(client(LOCAL_CLUSTER), ReindexAction.INSTANCE).source(sourceIndexInRemote)
+                .destination("desc-index-001")
+                .ignoreUnavailable(false)
```

Comment:
In contrast to this test, we want this case to succeed (i.e. not to throw an exception).

Reply:
Exactly. I'm trying to sort it our with search team. Will keep you posted

Reply:
this conversation was regarding test case
https://github.com/volodk85/elasticsearch/blob/2ae560e587506136e1dfc28f12124321f99b9c7d/modules/reindex/src/internalClusterTest/java/org/elasticsearch/index/reindex/CrossClusterReindexIT.java#L140-L166